pkg/acceptance/helpers/information_schema_client.go

```diff
@@ -25,20 +27,37 @@ func (c *InformationSchemaClient) client() *sdk.Client {
 	return c.context.client
 }
 
-func (c *InformationSchemaClient) GetQueryTextByQueryId(t *testing.T, queryId string) string {
+type QueryHistory struct {
+	QueryId   string
+	QueryText string
+	QueryTag  string
+}
+
+func (c *InformationSchemaClient) GetQueryHistory(t *testing.T, limit int) []QueryHistory {
 	t.Helper()
-	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => 20)) WHERE QUERY_ID = '%s'", queryId))
+	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT * FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => %d))", limit))
 	require.NoError(t, err)
-	require.Len(t, result, 1)
-	require.NotNil(t, result[0]["QUERY_TEXT"])
-	return (*result[0]["QUERY_TEXT"]).(string)
+	return collections.Map(result, func(m map[string]*any) QueryHistory {
+		require.NotNil(t, m["QUERY_ID"])
+		require.NotNil(t, m["QUERY_TEXT"])
+		require.NotNil(t, m["QUERY_TAG"])
+		return QueryHistory{
+			QueryId:   (*m["QUERY_ID"]).(string),
+			QueryText: (*m["QUERY_TEXT"]).(string),
+			QueryTag:  (*m["QUERY_TAG"]).(string),
+		}
+	})
 }
 
-func (c *InformationSchemaClient) GetQueryTagByQueryId(t *testing.T, queryId string) string {
+func (c *InformationSchemaClient) GetQueryHistoryByQueryId(t *testing.T, limit int, queryId string) QueryHistory {
 	t.Helper()
-	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT QUERY_TAG FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => 20)) WHERE QUERY_ID = '%s'", queryId))
+	result, err := c.client().QueryUnsafe(context.Background(), fmt.Sprintf("SELECT QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY(RESULT_LIMIT => %d)) WHERE QUERY_ID = '%s'", limit, queryId))
 	require.NoError(t, err)
 	require.Len(t, result, 1)
-	require.NotNil(t, result[0]["QUERY_TAG"])
-	return (*result[0]["QUERY_TAG"]).(string)
+	require.NotNil(t, result[0]["QUERY_TEXT"])
+	return QueryHistory{
+		QueryId:   (*result[0]["QUERY_ID"]).(string),
```

Comment:
1. Here we only select QUERY_TEXT, so why we set QueryTag here?
2. Why do we use a limit 20 and expect 1 result?
3. If we want to use the other columns, we should add `require.NotNil`, as above.

Reply:
Added common mapper that has `require.NotNil`, limit is on the number of queries we look through, so to find this one query by query id we have look through `limit` number of queries in query history

Reply:
Looks good now, I meant that we can use LIMIT 1 in GetQueryHistoryByQueryId, right? But that's a nit.